pkg/controller/jobs/job/job_controller.go

```diff
@@ -289,18 +287,14 @@ func (j *Job) Finished() (metav1.Condition, bool) {
 		}
 	}
 
-	condition := metav1.Condition{
-		Type:               kueue.WorkloadFinished,
-		Status:             metav1.ConditionTrue,
-		Reason:             "JobFinished",
-		Message:            "Job finished successfully",
-		ObservedGeneration: j.Generation,
-	}
+	success = true
+	message = "Job finished successfully"
```

Comment:
instead, copy the message from the Succeeded or Failed condition. Apply the same recommendation for all the other job types.

If somehow the conditions have empty messages, then we can put these placeholder messages.